repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/services/datastore/RemoteDatastoreServiceTest.groovy

```diff
@@ -21,142 +21,315 @@ package features.services.datastore
+
+        then: "the request is rate-limited"
+        messageChain.receivedResponse.code as Integer == SC_REQUEST_ENTITY_TOO_LARGE
+
+        and: "the request does not pass to the origin service"
+        messageChain.handlings.size() == 0
+    }
+
+    def "Repose instances should reconnect to the remote datastore when the remote datastore is restarted"() {
+        given: "three Repose instances are started, two to handle traffic, one to act as the remote datastore"
+        (repose1, repose1LogSearch) = startRepose('repose1', repose1Port, targetPort, datastorePort, true)
+        (repose2, repose2LogSearch) = startRepose('repose2', repose2Port, targetPort, datastorePort, true)
+        (remoteDatastore, reposeRemoteLogSearch) =
+            startRepose('remote', reposeRemotePort, targetPort, datastorePort, false)
+
+        and: "they are ready to service requests"
+        waitUntilReadyToServiceRequests(repose1LogSearch)
+        waitUntilReadyToServiceRequests(repose2LogSearch)
+        waitUntilReadyToServiceRequests(reposeRemoteLogSearch)
+
+        and: "requests will be made using the rate limiting group allowing 10 requests per hour"
+        def headers = ["X-PP-User": "user", "X-PP-Groups": "10_per_hour"]
+
+        when: "the user sends a request to each Repose instance"
+        def singleMessageChain1 = deproxy.makeRequest(url: repose1Endpoint, headers: headers)
+        def singleMessageChain2 = deproxy.makeRequest(url: repose2Endpoint, headers: headers)
+
+        then: "the requests are not rate-limited and pass to the origin service"
+        singleMessageChain1.receivedResponse.code as Integer == SC_OK
+        singleMessageChain2.receivedResponse.code as Integer == SC_OK
+        singleMessageChain1.handlings.size() == 1
+        singleMessageChain2.handlings.size() == 1
+
+        when: "the remote datastore is stopped"
+        remoteDatastore.stop()
+
+        and: "the user sends 10 requests to each Repose instance"
+        def manyMessageChains1 = (1..10).collect { deproxy.makeRequest(url: repose1Endpoint, headers: headers) }
+        def manyMessageChains2 = (1..10).collect { deproxy.makeRequest(url: repose2Endpoint, headers: headers) }
+
+        then: "all 20 requests are successful since they should be using their local datastore now"
+        manyMessageChains1.every { it.receivedResponse.code as Integer == SC_OK }
+        manyMessageChains2.every { it.receivedResponse.code as Integer == SC_OK }
+        manyMessageChains1.every { it.handlings.size() == 1 }
+        manyMessageChains2.every { it.handlings.size() == 1 }
+
+        when: "the remote datastore is started again and is ready to service requests"
+        (remoteDatastore, reposeRemoteLogSearch) =
+            startRepose('remote', reposeRemotePort, targetPort, datastorePort, false)
+        waitUntilReadyToServiceRequests(reposeRemoteLogSearch)
+
+        and: "the user sends 10 requests total (5 to each Repose instance)"
```

Comment:
Oh, are you referring to the fact that they would have hit their rate limit had the Remote Datastore not gone down? Yeah, from all of my testing, the Repose instances don't do anything to let the Remote Datastore know what's happened while it couldn't be reached, so essentially everything gets reset when:
1. the Remote Datastore goes down
2. the Remote Datastore comes back up